### DISCUSSION TITLE
Playbook impact on performance

### DISCUSSION THREAD

**Discussion by sblanch** [Security-Onion-Solutions/securityonion#6837] [Unsupported Versions] [answered]
Home Network (on-prem)
Intel i5 Processor (4 core)
Security Onion 2.3.91 Standard Install from ISO
1 Node
CentOS 7 Install
16GB RAM (unfortunately the max the system can hold)
1TB Traditional System Disk
1TB SSD /nsm
Avg 1.2Mb/s on Monitor Interface over 24 hours
Avg RAM usage (90%) over 24 hours
Avg CPU 12% over 24 hours
All services (so-status) green

I ran SO for several years at home (pre SO 2.0).  I finally gathered enough hardware pieces to meet the requirements for installing SO 2.x so I'm still learning the tuning via global.sls, etc.

I made the "critical' community plays active, but when I do, my load goes up to the point where the system is non-responsive (load >300, Grid shows fault if it loads at all) and I have to disable the span port on my switch which feeds the monitor port to SO and eventually the load returns to normal (< 1).  I have not enabled any experimental plays, so I didn't know it would affect the system as much as it did.  

Is this to be expected given the system (processor/RAM) I am running SO on?  
Is this likely the cause of my spike in load?  
How do they compare to the load placed by Playbook vs Suricata alert rules?
How do others determine which plays to enable on their SO boxes?

Any other insight on using Playbook is appreciated. 

Thanks,
Scott

**Comment by dougburks** [comment]
First, please make sure that you do NOT have the `Malicious Nishang PowerShell Commandlets` play enabled:
https://docs.securityonion.net/en/2.3/playbook.html#putting-a-play-into-production

**Comment by sblanch** [comment, marked as answer]
Yes, I had only enabled playbooks that had a level of critical.  Nishang has a level of high.  It was not enabled (status of Active).

Scott

**Reply by ccamejo1** [reply]
From what I have experienced on our deployment, any query done against the searchnodes causes a noticeable dip in ingestion and increase in our Redis queue.  I believe Playbooks use elastic queries to generate alerts based on the criteria specified in the play.  I'm not sure how many critical plays there are, but this could be what's causing the non-responsiveness you are seeing.
Our current deployment ingest approximately 4.1 billion events a day, and using Kibana for any search during peak hours causes a noticeable change in EPS for our forward nodes.  You may want to start by adding Plays one at a time and see how it affects performance vs adding all critical plays at once.
C²

**Reply by defensivedepth** [reply]
Yes, at its core, Playbook uses Elastalert, which queries Elasticsearch every 3 minutes. So if you enable 100 plays, that is 100 queries against elasticsearch every 3 minutes.

**Reply by sblanch** [reply]
Thanks.  This information is really helpful.  I'm continuing to learn how Playbook creates the .yaml files for alerting in the /opt/so/rules/elastalert/playbook directory.  I'll be more judicious in my activating the plays.  Glad I don't have 4.1 Billion events a day, but amazingly on my home network I do have about 1.7M.  And when Rumble.run does it discovery on Sundays it is significantly larger.   I appreciate this great community of knowlege.

Scott

**Comment by sblanch** [comment]
Some additional information.  The primary hard drive that the OS was installed on began acting flaky (my IOWait was off the chart some times).  I ran smartctl to see the state of the drive, and it was not good.  I have replaced it with a new SSD (copied old disk with dd) and all is running even better than before.  Sometimes it is hardware and not software causing the issues.
Scott

**Reply by defensivedepth** [reply]
Thanks for updating this with additional context - failing hardware will certainly screw things up!
